code-for-canada/project-thundercat
# [EditEmailActionDialog] Initial creation

# Description

Creation of Modify Email Response dialog.
This is a generic wrapper for the Add/Modify Email/Task components.

Also created `EditEmail` and `EditTask` to hold to specifics for modifying each response.

Clicking either 'Add Email Response' or 'Create a Task' will open this dialog. Clicking the 'Save Response' on the dialog button will update the email's state as before.

I did try to use PopupBox, but that would have required a great many additional parameters and a much larger overhaul of the entire application.

Two examples of using EditEmailActionDialog(as seen in Email.jsx):
```
<EditEmailActionDialog
          showDialog={this.state.showAddEmailDialog}
          handleClose={this.closeEmailDialog}
          saveEmail={this.replyToEmail}
          actionType={ACTION_TYPE.email}
          editMode={EDIT_MODE.create}
        />
<EditEmailActionDialog
          showDialog={this.state.showAddTaskDialog}
          handleClose={this.closeTaskDialog}
          saveEmail={this.addTaskToEmail}
          actionType={ACTION_TYPE.task}
          editMode={EDIT_MODE.create}
        />
```

- showDialog - the state variable that is used to determine if the modal is hidden or visible
- handleClose - the function to call when closing the modal (saving or just exiting)
- saveEmail - the function to call when saving the response (handleClose AND saveEmail will be called when the button is triggered)
- actionType - is this an email or a task
- editMode - is this a new item (add), or the modification of an old one.


## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Screenshot

![image](https://user-images.githubusercontent.com/2746350/55552242-e51fb400-56aa-11e9-9f2c-10458aaa47ab.png)

Clicking 'Add Email Response'
![image](https://user-images.githubusercontent.com/4640747/55579348-02be3f00-56e6-11e9-9014-5313a0e40f17.png)

Closing the dialog without saving
![image](https://user-images.githubusercontent.com/2746350/55552318-0e404480-56ab-11e9-8503-b966762cb3d6.png)

Clicking 'Create a task'
![image](https://user-images.githubusercontent.com/4640747/55579370-0f429780-56e6-11e9-9592-035a9476e4b6.png)


Saving the response
![image](https://user-images.githubusercontent.com/2746350/55552400-4182d380-56ab-11e9-934f-bfde23666510.png)



# Testing

Applicable for all code changes.

Manual steps to reproduce this functionality:

1.  Go to Prototype -> Start eMIB -> Step 5 -> Start Test
2.  Open the 'Inbox' Tab
3.  Click 'Add email response', verify that the dialog appears
4.  Close the dialog without saving
5.  Verify that the responded to icon does not appear
6.  Click 'Add email response' again, and click 'Save response'
7. Open another email and repeat steps 3-7, but with the 'Create a task' button this time

# Checklist

Applicable for all code changes.
